src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -0,0 +1,289 @@
+import {
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType,
+    RequestError
+} from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import isVaultedInstrument from '../../is-vaulted-instrument';
+import Payment, { HostedInstrument } from '../../payment';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import {
+    AdyenCardState,
+    AdyenCheckout,
+    AdyenComponent,
+    AdyenConfiguration,
+    AdyenError,
+    ResultCode,
+    ThreeDS2ComponentType,
+    ThreeDS2OnComplete,
+    ThreeDS2Result,
+} from './adyenv2';
+import AdyenV2PaymentInitializeOptions from './adyenv2-initialize-options';
+import AdyenV2ScriptLoader from './adyenv2-script-loader';
+
+export default class AdyenV2PaymentStrategy implements PaymentStrategy {
+    private _adyenCheckout?: AdyenCheckout;
+    private _stateContainer?: string;
+    private _adyenv2?: AdyenV2PaymentInitializeOptions;
+    private _adyenComponent?: AdyenComponent;
+    private _adyenOldComponent?: AdyenComponent;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _adyenV2ScriptLoader: AdyenV2ScriptLoader,
+        private _formPoster: FormPoster
+
+    ) {}
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { adyenv2 } = options;
+
+        if (!adyenv2) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.adyenv2" argument is not provided.');
+        }
+
+        const paymentMethod = this._store.getState().paymentMethods.getPaymentMethod(options.methodId);
+
+        if (!paymentMethod) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        this._adyenv2 = adyenv2;
+
+        const configuration: AdyenConfiguration = {
+            environment:  paymentMethod.initializationData.environment,
+            locale: this._getLocale(),
+            originKey: paymentMethod.initializationData.originKey,
+            paymentMethodsResponse: paymentMethod.initializationData.paymentMethodsResponse,
+        };
+
+        return this._adyenV2ScriptLoader.load(configuration)
+            .then(adyenCheckout => {
+                this._adyenCheckout = adyenCheckout;
+
+                const adyenComponent = this._adyenCheckout.create(paymentMethod.method, {
+                        ...adyenv2.options,
+                        onChange: (state: AdyenCardState) => {
```

Comment:
If you allow the client app to be able to override every option Adyen JS allows, what you have won't work properly as it will override the `onChange` option that's passed in. What you need to do is to call `options.onChange` (if it exists) here so it doesn't get completely overridden. Otherwise, if you don't want the client app to be able to specify an `onChange` callback, you'll need to omit it from the public interface.

Reply:
I'll omit onChange callback, since the only behavior of this is to get the data from adyen component so it is not needed to override